worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/clipboard/DiskOptimizedClipboard.java

```diff
@@ -127,15 +152,80 @@ public DiskOptimizedClipboard(BlockVector3 dimensions, File file) {
         }
     }
 
+    /**
+     * Load an existing file as a DiskOptimizedClipboard. The file MUST exist and MUST be created as a DiskOptimizedClipboard
+     * with data written to it.
+     */
     public DiskOptimizedClipboard(File file) {
-        super(readSize(file), BlockVector3.ZERO);
+        this(file, VERSION);
+    }
+
+    /**
+     * Load an existing file as a DiskOptimizedClipboard. The file MUST exist and MUST be created as a DiskOptimizedClipboard
+     * with data written to it.
+     *
+     * @param file            File to read from
+     * @param versionOverride An override version to allow loading of older clipboards if required
+     */
+    public DiskOptimizedClipboard(File file, int versionOverride) {
+        super(readSize(file, versionOverride), BlockVector3.ZERO);
+        headerSize = getHeaderSizeOverrideFromVersion(versionOverride);
         nbtMap = new HashMap<>();
         try {
             this.file = file;
             this.braf = new RandomAccessFile(file, "rw");
             braf.setLength(file.length());
+            this.nbtBytesRemaining = Integer.MAX_VALUE - (int) file.length();
             init();
-            if (braf.length() - HEADER_SIZE == ((long) getVolume() << 1) + (long) ((getHeight() >> 2) + 1) * ((getLength() >> 2) + 1) * ((getWidth() >> 2) + 1)) {
+            long length = braf.length();
+            long biomeLength = (long) ((getHeight() >> 2) + 1) * ((getLength() >> 2) + 1) * ((getWidth() >> 2) + 1);
+            if (headerSize >= VERSION_2_HEADER_SIZE) {
+                readBiomeStatusFromHeader();
+                int nbtCount = readNBTSavedCountFromHeader();
+                int entitiesCount = readEntitiesSavedCountFromHeader();
+                if (Settings.settings().CLIPBOARD.SAVE_CLIPBOARD_NBT_TO_DISK && nbtCount + entitiesCount > 0) {
+                    int biomeBlocksLength = (int) (headerSize + (getVolume() << 1) + (hasBiomes ? biomeLength : 0));
```

Comment:
Could this be moved to a separate method? That would make the code a bit cleaner imo.